src/Java.Interop/Java.Interop/JniTypeSignature.cs

```diff
@@ -39,7 +39,7 @@ public struct JniTypeSignature : IEquatable<JniTypeSignature>
 			get {return ArrayRank == 0 ? SimpleReference ?? throw new InvalidOperationException (): QualifiedReference;}
 		}
 
-		public JniTypeSignature (string simpleReference, int arrayRank = 0, bool keyword = false)
+		public JniTypeSignature (string? simpleReference, int arrayRank = 0, bool keyword = false)
```

Comment:
Why do we allow a null value here?  We don't even test what happens when `null` is provided!

Does it even make sense to allow `null` here?

Reply:
Oh, it's a struct; there's no way to prevent `null` from here.

Reply:
That said, that doesn't in turn mean that *this* constructor needs to support a `null` parameter, just that it's not possible *in general* to *prevent* `SimpleReference` from being `null`.

Thoughts?

Reply:
That said, that doesn't in turn mean that *this* constructor needs to support a `null` parameter, just that it's not possible *in general* to *prevent* `SimpleReference` from being `null`.

Thoughts?

Reply:
I can see just instance when `SimpleReference` should be allowed to be null - creation of `JniTypeSignature.Empty` field to use as the default value for variables. As you mentioned on Slack, though, this could be done using a default constructor (possibly private?), though. Could `SimpleReference` be `String.Empty` by default?

Reply:
> Could SimpleReference be `String.Empty` by default

No.  It's a `struct`; `struct`s are special, and one way in which they're special is that they *always* have a default constructor which "zero-initializes" all members, which means that after invoking the default constructor `SimpleReference` can *never* be `string.Empty`; it can *only* be `null`.

Which is in part why `JniTypeSignature.SimpleReference` is of type `string?` (after enabling nullable-reference-types); there's always a chance it *could* be null, so all callers need to be careful.

That does not in any way mean or imply that the *constructor* needs to allow `simpleReference` to be `null`.  It *may* be "nice" in that it allows "equivalent construction" between the default constructor and the non-default constructor -- which in itself may be a *good* thing! -- but it isn't *required*.

I can't find anything in the Framework Design Guidelines which address this point.